opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/metrics/MetricAdapter.java

```diff
@@ -0,0 +1,363 @@
+      for (Point point : ts.getPoints()) {
+        DoubleSummaryPointData otelPoint =
+            point
+                .getValue()
+                .match(
+                    dv -> null,
+                    lv -> null,
+                    distribution -> null,
+                    summary ->
+                        DoubleSummaryPointData.create(
+                            startTimestamp,
+                            mapTimestamp(point.getTimestamp()),
+                            attributes,
+                            summary.getCount(),
+                            summary.getSum(),
+                            mapValueAtPercentiles(summary.getSnapshot().getValueAtPercentiles())),
+                    defaultValue -> null);
+        if (otelPoint != null) {
+          result.add(otelPoint);
+        }
+      }
+    }
+    return result;
+  }
+
+  static Attributes mapAttributes(List<LabelKey> labels, List<LabelValue> values) {
+    AttributesBuilder result = Attributes.builder();
+    for (int i = 0; i < labels.size(); i++) {
+      result.put(labels.get(i).getKey(), values.get(i).getValue());
+    }
+    return result.build();
+  }
+
+  static long longValue(Point point) {
+    return point
+        .getValue()
+        .match(
+            Double::longValue,
+            lv -> lv,
+            // Ignore these cases (logic error)
+            distribution -> 0,
+            summary -> 0,
+            defaultValue -> 0)
+        .longValue();
+  }
+
+  static double doubleValue(Point point) {
+    return point
+        .getValue()
+        .match(
+            d -> d,
+            Long::doubleValue,
+            // Ignore these cases (logic error)
+            distribution -> 0,
+            summary -> 0,
+            defaultValue -> 0)
+        .doubleValue();
+  }
+
+  static List<Double> mapBoundaries(Distribution.BucketOptions censusBuckets) {
+    return censusBuckets.match(
+        explicit -> explicit.getBucketBoundaries(), defaultOption -> Collections.emptyList());
+  }
+
+  static List<Long> mapCounts(List<Distribution.Bucket> buckets) {
+    List<Long> result = new ArrayList<>(buckets.size());
+    for (Distribution.Bucket bucket : buckets) {
+      result.add(bucket.getCount());
+    }
+    return result;
+  }
+
+  static List<ExemplarData> mapExemplars(List<Distribution.Bucket> buckets) {
+    List<ExemplarData> result = new ArrayList<>();
+    for (Distribution.Bucket bucket : buckets) {
+      Exemplar exemplar = bucket.getExemplar();
+      if (exemplar != null) {
+        result.add(mapExemplar(exemplar));
+      }
+    }
+    return result;
+  }
+
+  private static ExemplarData mapExemplar(Exemplar exemplar) {
+    // Look for trace/span id.
+    String spanId = null;
+    String traceId = null;
+    if (exemplar.getAttachments().containsKey("SpanContext")) {
+      // We need to use `io.opencensus.contrib.exemplar.util.AttachmentValueSpanContext`
+      // The `toString` will be the following:
```

Comment:
@bogdandrutu ;)

Reply:
As an FYI - I created a test directly against the exemplar-utils library, but still did not take a dependency on it in the shim.